cloud/ec2.go

```diff
@@ -868,15 +868,29 @@ func (m *ec2Manager) TerminateInstance(ctx context.Context, h *host.Host, user,
 		return errors.Wrap(h.Terminate(ctx, user, fmt.Sprintf("detected invalid instance ID '%s'", h.Id)), "terminating instance in DB")
 	}
 
-	if h.NoExpiration {
-		// Clean up remaining DNS records for unexpirable hosts.
-		grip.Error(message.WrapError(deleteHostPersistentDNSName(ctx, m.env, h, m.client), message.Fields{
+	// Any host that has been unexpirable will have been given a DNS name, which we need to clean up.
+	if h.PersistentDNSName != "" {
+		dnsName := h.PersistentDNSName
+		err := deleteHostPersistentDNSName(ctx, m.env, h, m.client)
+		if err == nil {
```

Comment:
nit: for style consistency pref to log out the err if it is not nil and then the success message afterwards